src/spatio_temporal_voxel_layer.cpp

```diff
@@ -768,6 +776,39 @@ void SpatioTemporalVoxelLayer::SaveGridCallback(
   resp->status = false;
 }
 
+/*****************************************************************************/
+void SpatioTemporalVoxelLayer::ClearGridExceptRegionCallback(
+  const std::shared_ptr<rmw_request_id_t>/*request_header*/,
+  const std::shared_ptr<nav2_msgs::srv::ClearCostmapExceptRegion::Request> request,
+  std::shared_ptr<nav2_msgs::srv::ClearCostmapExceptRegion::Response> response)
+/*****************************************************************************/
+{
+  if (_mapping_mode) {
+    RCLCPP_WARN(node_->get_logger(),
+      "Received request to clear the grid while on mapping mode. Not doing anything.");
+    return;
+  }
+
+  RCLCPP_INFO(node_->get_logger(),
+    "Received request to clear the grid except a square region of side " + std::to_string(request->reset_distance));
+
+  _clear_grid_except_region = true;
+  _reset_distance = request->reset_distance;
+}
+
+/*****************************************************************************/
+void SpatioTemporalVoxelLayer::ClearGridExceptRegion(
+  double robot_x, double robot_y,
+  std::unordered_set<volume_grid::occupany_cell> & cleared_cells)
+/*****************************************************************************/
+{
+  if (!_clear_grid_except_region)
+    return;
```

Comment:
Add brackets in the if condition, its a 1 liners so I know you "can" but if we expanded that later I've had problems where developers forget to later add it

Reply:
Right, that's understandable. Added the brackets there. 

Yes have been testing and re-testing this after the changes!